### ISSUE TITLE
Unable to put() with any data

### ISSUE THREAD

**Issue by ianchadwick** [mailgun/mailgun-php#122]
Attempting the below I get "Invalid resource type: array".

```php
$mailgun->put('domains/' . $domain . '/connection', [
    'require_tls' => true,
    'skip_verification' => true,
]);
```

The only way I've been able to update the item is to use 'domains/' . $domain . '/connection?require_tls=true' which is a bit hacky.

Also if you don't pass anything as the 2nd param it complains due to missing argument.

**Comment by nisbeti** [comment]
Since the upgrade to v2 I have the same problem, which I have managed to (hack) fix, until a proper fix is released. I am using 
```php
$client = new \Http\Adapter\Guzzle6\Client();
```
as I didn't want to go the Puli discovery route.

./vendor/guzzlehttp/psr7/src/functions.php has a function called ```stream_for``` at line 77, which checks the type of the $resource passed to it (in our case this is the $body of our PUT request), and rejects it if it is an array.

Either we must send the $body as a urlencoded string, ```'name=Bob&subscribed=false'``` along with setting a content-type header of ```application/x-www-form-urlencoded``` (which means hacking the mailgun RestClient for now), or we can stick with the array of params, but then we still have to hack to RestClient in order to convert the array into a urlencoded string and set the header.

Here is the hack for file ```./vendor/mailgun/mailgun-php/src/Mailgun/Connection/RestClient.php```

*This is Strictly Temporary, as it will be overwritten next time mailgun-php is updated*

At line 86 (just before ```$request = new Request($method, $this->getApiUrl($uri), $headers, $body);``` insert:
```php
        if (is_array($body)) {
            $headers['Content-Type'] = 'application/x-www-form-urlencoded';
            $urlencodedBody = '';
            foreach ($body as $key => $value) {
                $urlencodedBody .= urlencode($key) . '=' . urlencode($value) . '&';
            }
            $body = trim($urlencodedBody,'&');
        }
```

A more permanent fix may include being able to send the content-type header from the PUT method, but I am unsure what impact that will have when not using the Guzzle6\Client and using Puli discover instead.

**Comment by iRbouh** [comment]
First, make sure you are using the last version of `php-http/guzzle6-adapter`, then try the following:

```
$client = \Http\Adapter\Guzzle6\Client::createWithConfig([
            'headers' => [
                'Content-Type' => 'application/x-www-form-urlencoded'
            ]
        ]);
$mgClient = new \Mailgun\Mailgun('YOUR_API_KEY', $client);

$listAddress = 'LIST@YOUR_DOMAIN_NAME';
$memberAddress = 'bob@example.com';

# Issue the call to the client.
$result = $mgClient->put("lists/$listAddress/members/$memberAddress", http_build_query([
    'subscribed' => 'no',
    'name'       => 'Foo Bar'
]));
```

**Comment by matijakovacevic** [comment]
@iRbouh you're solution works perfectly!

To me, it's much more cleaner than that of @nisbeti, due to the fact that it's the $client's responsibility to add custom headers. Plus, works even when Mailgun-php is updated.

**Comment by nisbeti** [comment]
Yes, much more elegant, and it avoids hacking. The examples in the documentation (https://documentation.mailgun.com/api-mailinglists.html#examples) should now be updated to reflect that an array cannot be passed as the second argument to the ```put``` method, and that the array should be wrapped in http_build_query().

**Comment by Nyholm** [comment]
Closed by #138
